sdk/typescript/src/eth-light-client/eth-light-client.ts

```diff
@@ -0,0 +1,305 @@
+ * 3. **Initialize State**: Uses the bootstrap data to initialize the Ethereum light client state in BCS format.
+ * 4. **Fetch Updates**: Retrieves updates from the Ethereum consensus RPC since the initial sync period.
+ * 5. **Prepare Transaction**: Constructs a transaction to call the `init_state` function in the Ethereum state module,
+ *    providing the necessary arguments such as the state bytes, network, contract address, and updates.
+ * 6. **Execute Transaction**: Signs and executes the transaction to initialize the Ethereum state on the dWallet network.
+ *
+ * **Arguments**
+ * @param {string} network - The Ethereum network to initialize (e.g., 'mainnet' or 'holesky').
+ * @param {string} rpc - The Ethereum consensus RPC endpoint.
+ * @param {string} contractAddress - The address of the Ethereum smart contract.
+ * @param {number} contractApprovedTxSlot - The slot of the data structure that holds approved transactions in the Ethereum smart contract.
+ * @param {Keypair} keypair - The keypair used to sign the transaction.
+ * @param {DWalletClient} client - The dWallet client instance.
+ */
+export const initEthereumState = async (
+	network: string,
+	rpc: string,
+	contractAddress: string,
+	contractApprovedTxSlot: number,
+	keypair: Keypair,
+	client: DWalletClient,
+) => {
+	let checkpoint = '';
+	switch (network) {
+		case 'mainnet': {
+			checkpoint = '0x8bfa089414dc5fe78dadc8b160a097fe744f17a80251f08eed0a3cdcc60b42f4';
+			break;
+		}
+		case 'holesky': {
+			checkpoint = '0x12e6b81891d23c90502dbc2354de9cb52afe4ff823ca00fd41d411213c6e7bbb';
+			break;
+		}
+		default: {
+			throw new Error('Invalid network');
+		}
+	}
+	let bootstrapJson = await getBootstrapData(rpc, checkpoint);
+	let bootstrap = bootstrapJson['data'];
+
+	let state = get_initial_state_bcs(checkpoint, rpc, network, bootstrap);
+	let stateBytes: Uint8Array = state['bytes'];
+	let syncPeriod = get_current_period(stateBytes);
+
+	let updatesResponseJson = await getUpdates(rpc, syncPeriod);
+	let updatesJson = JSON.stringify(updatesResponseJson.map((update: any) => update['data']));
+	let updatesBcs = stringToArrayU8Bcs(updatesJson);
+
+	let finalityUpdateResponse = await getFinalityUpdate(rpc);
+	let finalityUpdateJson = JSON.stringify(finalityUpdateResponse['data']);
+	let finalityUpdateBcs = stringToArrayU8Bcs(finalityUpdateJson);
+
+	let optimisticUpdateResponse = await getOptimisticUpdate(rpc);
+	let optimisticUpdateJson = JSON.stringify(optimisticUpdateResponse['data']);
+	let optimisticUpdateBcs = stringToArrayU8Bcs(optimisticUpdateJson);
+
+	let stateBcs = bcs.vector(bcs.u8()).serialize(stateBytes, {
+		size: stateBytes.length,
+		maxSize: stateBytes.length * 2,
+		allocateSize: stateBytes.length,
+	});
+
+	let contractAddressArrayU8 = ethers.getBytes(contractAddress);
+	let contractAddressBcs = bcs.vector(bcs.u8()).serialize(contractAddressArrayU8);
+
+	const tx = new TransactionBlock();
+	tx.moveCall({
```

Comment:
IIUC, this light client is keep being update constantly after this in the background by the validators right?

Reply:
No, the state is only updated upon calling `verify_new_state` function, that verifies the updates that are sent by the users and applies them to the state if they are valid.